### ISSUE TITLE
Thumbnails are not deterministic

### ISSUE THREAD

**Issue by Tisten** [OurMachinery/themachinery-public#1080]
- **Version:** Git Master Branch
- **OS:** Windows

While the fixes to https://github.com/OurMachinery/themachinery-public/issues/980 improved the situation a lot, there is still a lot of randomness in the thumbnail generation, coming from post effects (at least exposure) but also settings like clicking on the "use lighting environment asset". It would be nice if the thumbnails were rendered in a deterministic way.

**Comment by tobias-ourmachinery** [comment]
I assume you are talking about entity thumbnails now? 

Thumbnails for textures and materials shouldn't be regenerated unless there's an actual change to the creation graph representing them in which case determinism in the output matters less. 

In general I do agree that we should strive for deterministic output from the thumbnail generator but it can be hard to achieve in practice as there can be lots of temporal effects in play. With that said, we are working on getting some basic automated regression testing of the rendered output (image comparison) and in that scenario determinism is obviously very important -- we'll see how far we manage to push that and if we can use some of that work to make thumbnail gen fully deterministic.

**Comment by Tisten** [comment]
Yes, entity thumbnails. And my preferred solution would be to simply turn off post effects, or to always reset the initial start state so you don't start with a random jitter pattern for TAA or rely on last frame's exposure.

**Comment by niklas-ourmachinery** [comment]
I think the main decision is if we want the entity thumbnails to be:

* Completely machine generated in a deterministic way (could be implemented as a batch process).
* Hand-picked by users to show off the entity in a good way.

There are advantages and disadvantages to each. It's nice to be able to generate all thumbnails offline in a deterministic way, but it might not always be able to get a "good" picture of the entity.

One option is to use some kind of combination... we use the auto-generated thumbnail, unless the a user has chosen to take a manual thumbnail screenshot for a particular enitty.

**Comment by Tisten** [comment]
Letting the user manually generate a thumbnail would be great, but then it shouldn't be overwritten by an autogenerated thumbnail directly afterwards. My main point with the issue is that autogenegated so as thumbnails data should not cause version control conflicts. And right now there can be two unrelated changes to an entity which will merge just fine, but the thumbnail (which might not need to change in reality) will conflict because of temporal randomness.
